### PR TITLE
feat(docs): add integrations and examples

### PR DESCRIPTION
**Pull request by aranw**
Called the section integrations but I guess this could be named something else maybe?

I added the integrations section as 12 so that I wasn't renumbering everything and then possibly conflicting with other contributions

Closes https://github.com/a-h/templ/issues/50

### CODE REVIEW

File: examples/template/html.templ [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package template
```
Comment (a-h) [inline examples/template/html.templ:1]: I think this is an accidental commit?

File: examples/template/html_templ.go
Comment (a-h) [inline examples/template/html_templ.go:1]: Same with this?

**Comment by a-h** [comment]
This looks great, thanks for doing that! Just checking whether you meant to commit one of the files.

**Comment by aranw** [comment]
> This looks great, thanks for doing that! Just checking whether you meant to commit one of the files.

Yeah that was a mistake! I've removed it now

**Review by a-h — APPROVED** [review]

**Comment by a-h** [comment]
Thanks a lot!

**Comment by joerdav** [comment]
Odd that the PR checks didn't run on this, it looks like the examples were generated with a non-HEAD version of templ.

Maybe we need a contribution guide that says you should run `go run ./cmd/templ generate` rather than `templ generate` before commiting.

**Pull request merged**